src/main/java/upm/softwaredesign/finalproject/service/ActorService.java

```diff
@@ -19,20 +21,45 @@ public ActorService(ActorRepository repository) {
 		this.repository = repository;
 	}
 
-
-	public Optional<ActorEntity> findById(Integer id ) {
-		return this.repository.findById(id);
+	public Actor cast(ActorEntity actorEntity)
+	{
+		Actor actor = new Actor();
+		actor.setName(actorEntity.getName());
+		actor.setType(actorEntity.getType());
+		return actor;
+	}
+	public Actor findByType(String type ) {
+		List<ActorEntity> actors = this.repository.findAll();
+		for (int i = 0; i < actors.size(); i++) {
+			if(actors.get(i).getType().equals(type))
+			{
+				return this.cast(actors.get(i));
```

Comment:
This will return only the first Actor with the wanted type, and not all, is that what you want to do? Maybe if you want to return all actors of the specified type, a list of actors should be returned.